src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -217,9 +220,6 @@ public void fetchParticipationsWithSubmissionsAndResultsForCourses(List<Course>
      */
     public Course findOneWithExercisesAndLecturesAndExamsAndLearningGoalsAndTutorialGroupsForUser(Long courseId, User user, boolean refresh) {
         Course course = courseRepository.findByIdWithLecturesElseThrow(courseId);
-        if (!authCheckService.isAtLeastStudentInCourse(course, user)) {
```

Comment:
Can you please double-check that all other calls to this method do some authorization check before that? We should not leak the exercises to unregistered users.

Reply:
Its only usage is in the `/for-dashboard` endpoint 👍